server/common/transport-shared/src/main/java/org/kaaproject/kaa/server/transport/RangeExpressionParser.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2014-2016 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.server.transport;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Converts a range expression ("1-3, 2-7, 15") into a sequence of numbers from that range
```

Comment:
Add example as 1-3 => 1,2,3, and note that it leaves only unique values, i. e. (1-3, 2-5) => (1,2,3) v (2,3,4,5) => (1,2,3,4,5)